lib/Runtime/Library/DataView.cpp

```diff
@@ -360,12 +360,12 @@ namespace Js
         {
             JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedDataView);
         }
-        if (args.Info.Count < 3)
+        if (args.Info.Count < 2)
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_DataView_NeedArgument,  _u("offset or value"));
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_DataView_NeedArgument,  _u("offset"));
         }
         DataView* dataView = VarTo<DataView>(args[0]);
-        int8 value = JavascriptConversion::ToInt8(args[2], scriptContext);
+        int8 value = args.Info.Count == 2 ? 0 : JavascriptConversion::ToInt8(args[2], scriptContext);
```

Comment:
The logic looks correct to me, BUT, looking at this several branches are taken, imagine the "normal" case where no error will be thrown, we do:

1. if arg0 (i.e. `this`) doesn't exist throw
2. if arg0 is not a dataView, throw
3. if there are less than 2 args (i.e. if arg1 doesn't exist), throw
4. if there is no 3rd arg, i.e. follow the default to 0/NaN case
5. only if none of those conditions were met do the "normal" version

Could we optimise this slightly?

Minimally the args == 0 check can be combined with the < 2 check

More extensively we could do:
1. define the value variable at the start without initialising.
2. check for less than 3 args
3. within the `false` path, do the VarIs<DataView> check but no others, set the value to the conversion result
4. within the `true` path  do all the other checks a) less than 2 args = throw b) arg0 not a DataView = throw, otherwise set the value to 0 (or Nan)

Reply:
Done.